File: SDKs/amazonaws.com/iot/2015-05-28/python/src/sdk/models/shared/audittaskmetadata.py
import dataclasses
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import audittaskstatus_enum as shared_audittaskstatus_enum
from ..shared import audittasktype_enum as shared_audittasktype_enum


@dataclass_json
@dataclasses.dataclass
class AuditTaskMetadata:
    r"""AuditTaskMetadata
    The audits that were performed.
    """
    
    task_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('taskId') }})
    task_status: Optional[shared_audittaskstatus_enum.AuditTaskStatusEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('taskStatus') }})
    task_type: Optional[shared_audittasktype_enum.AuditTaskTypeEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('taskType') }})
    

File: SDKs/amazonaws.com/iot/2015-05-28/java/lib/src/main/java/openapisdk/models/shared/AuditTaskMetadata.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * AuditTaskMetadata
 * The audits that were performed.
**/
public class AuditTaskMetadata {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("taskId")
    public String taskId;
    public AuditTaskMetadata withTaskId(String taskId) {
        this.taskId = taskId;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("taskStatus")
    public AuditTaskStatusEnum taskStatus;
    public AuditTaskMetadata withTaskStatus(AuditTaskStatusEnum taskStatus) {
        this.taskStatus = taskStatus;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("taskType")
    public AuditTaskTypeEnum taskType;
    public AuditTaskMetadata withTaskType(AuditTaskTypeEnum taskType) {
        this.taskType = taskType;
        return this;
    }
}